integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/PartitionDropper.scala

```diff
@@ -0,0 +1,130 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.carbondata.spark.rdd
+
+import java.io.IOException
+
+import org.apache.spark.sql.execution.command.DropPartitionCallableModel
+import org.apache.spark.util.PartitionUtils
+
+import org.apache.carbondata.common.logging.LogServiceFactory
+import org.apache.carbondata.core.metadata.schema.partition.PartitionType
+import org.apache.carbondata.spark.{AlterPartitionResultImpl, PartitionFactory}
+
+object PartitionDropper {
+
+  val logger = LogServiceFactory.getLogService(PartitionDropper.getClass.getName)
+
+  def triggerPartitionDrop(dropPartitionCallableModel: DropPartitionCallableModel): Unit = {
+    val sc = dropPartitionCallableModel.sqlContext.sparkContext
```

Comment:
better to remove sc from DropPartitionCallableModel.
suggest to package the parameters of AlterTableLoadPartitionRDD and CarbonScanPartitionRDD